packages/react-router/lib/dom/ssr/browser.tsx

```diff
@@ -168,6 +209,24 @@ function createHydratedRouter(): RemixRouter {
     }
 
     if (hydrationData && hydrationData.errors) {
+      // If we rendered a 404 during SSR but then matched a prop route on the
+      // client, we want to clear out the 404 and let the client route render.
+      // This will cause a flicker/hydration error but will only happen in some
+      // edge cases:
+      // - During migrations from RouterProvider -> Vite Plugin
+      // - If the app is using SSR
+      // - If users actually enter the site on a client-only URL
+      if (matchedPropRoute) {
+        for (let [routeId, error] of Object.entries(hydrationData.errors)) {
+          if (isRouteErrorResponse(error) && error.status === 404) {
+            delete hydrationData.errors[routeId];
```

Comment:
Wipe out any SSR'd 404s if we now match our new client routes - we will have a hydration error/flicker but it will recover with the client render